src/logic/safe/store/actions/createTransaction.ts

```diff
@@ -91,231 +62,86 @@ const getSafeTxGas = async (txProps: RequiredTxProps, safeVersion: string): Prom
-
-    logError(Errors._803, err.message)
-
-    errorCallback?.()
-
-    notifications.closePending()
-
-    // Existing transaction was being finalised (txId exists)
-    if (isFinalization && txId) {
-      dispatch(removePendingTransaction({ id: txId }))
-    }
-
-    const executeDataUsedSignatures = safeInstance.methods
-      .execTransaction(
-        txProps.to,
-        txProps.valueInWei,
-        txProps.txData,
-        txProps.operation,
-        0,
-        0,
-        0,
-        ZERO_ADDRESS,
-        ZERO_ADDRESS,
-        txArgs.sigs,
-      )
-      .encodeABI()
-    const contractErrorMessage = await fetchOnchainError(executeDataUsedSignatures, safeInstance, from)
-
-    notifications.showOnError(err, contractErrorMessage)
-  }
-
-  async onlyConfirm(hardwareWallet: boolean): Promise<string | undefined> {
-    const { txArgs, safeTxHash, txProps, safeVersion } = this
-
-    return await tryOffChainSigning(
-      safeTxHash,
-      { ...txArgs, sender: String(txArgs.sender), safeAddress: txProps.safeAddress },
-      hardwareWallet,
-      safeVersion,
-    )
-  }
-
-  async sendTx(): Promise<string> {
-    const { txArgs, isFinalization, from, safeTxHash, txProps, dispatch, txId } = this
-
-    const tx = isFinalization ? getExecutionTransaction(txArgs) : getApprovalTransaction(this.safeInstance, safeTxHash)
-    const sendParams = createSendParams(from, txProps.ethParameters || {})
-    const promiEvent = tx.send(sendParams)
-
-    // When signing on-chain don't mark as pending as it is never removed
-    if (isFinalization) {
-      // Finalising existing transaction (txId exists)
-      if (txId) {
-        dispatch(addPendingTransaction({ id: txId }))
-      }
-      aboutToExecuteTx.setNonce(txArgs.nonce)
-    }
-
-    return new Promise((resolve, reject) => {
-      promiEvent.once('transactionHash', resolve) // this happens much faster than receipt
-      promiEvent.once('error', reject)
-    })
-  }
-
-  async submitTx(
-    state: AppReduxState,
-    confirmCallback?: ConfirmEventHandler,
-    errorCallback?: ErrorEventHandler,
-  ): Promise<void> {
-    const { isFinalization, safeVersion } = this
-    const { hardwareWallet, smartContractWallet } = providerSelector(state)
-    const canSignOffChain = checkIfOffChainSignatureIsPossible(isFinalization, smartContractWallet, safeVersion)
-    // Off-chain signature
-    if (!isFinalization && canSignOffChain) {
-      try {
-        const signature = await this.onlyConfirm(hardwareWallet)
-        this.onComplete(signature, confirmCallback)
-      } catch (err) {
-        // User likely rejected transaction
-        logError(Errors._814, err.message)
-      }
-      return
-    }
-
-    // On-chain signature or execution
-    try {
-      await this.sendTx()
-      this.onComplete(undefined, confirmCallback)
-    } catch (err) {
-      this.onError(err, errorCallback)
-    }
+// Assign defaults to txArgs
+const getTxCreationTxArgs = async (
```

Comment:
Any reason why this is not named `getTxCreationArgs`?

Reply:
Somewhat confusing but as per @mikheevm's suggestion, this follows something like this: get transaction creation `txArgs` aka "get the `txArgs` of the transaction being created".